flowzone.yml

```diff
@@ -1529,87 +1535,39 @@ jobs:
 
-      - name: Set up Python 3.7
-        if: steps.python_poetry.outputs.enabled == 'true'
-        uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1 # v4
-        with:
-          python-version: "3.7"
-
-      - name: Check compatibility
-        if: steps.python_poetry.outputs.enabled == 'true'
-        run: |
-          error_check=`(poetry env use 3.7 2>&1 || true)`
-          if ! grep -q "Please choose a compatible version" <<< $error_check
-          then
-            echo "PYTHON_VERSIONS=$(echo "${PYTHON_VERSIONS}" | jq -c '. + ["3.7"]')" >> $GITHUB_ENV
-          else
-            echo "Python 3.7 does not meet project requirements."
-          fi
-
-      - name: Set up Python 3.8
+      - name: Set up Python environment
         if: steps.python_poetry.outputs.enabled == 'true'
         uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1 # v4
         with:
-          python-version: "3.8"
+          python-version: ${{ env.SUPPORTED_VERSIONS }}
 
-      - name: Check compatibility
+      - name: Validate project Python requirements
         if: steps.python_poetry.outputs.enabled == 'true'
         run: |
-          error_check=`(poetry env use 3.8 2>&1 || true)`
-          if ! grep -q "Please choose a compatible version" <<< $error_check
-          then
-            echo "PYTHON_VERSIONS=$(echo "${PYTHON_VERSIONS}" | jq -c '. + ["3.8"]')" >> $GITHUB_ENV
-          else
-            echo "Python 3.8 does not meet project requirements."
-          fi
-
-      - name: Set up Python 3.9
-        if: steps.python_poetry.outputs.enabled == 'true'
-        uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1 # v4
-        with:
-          python-version: "3.9"
-
-      - name: Check compatibility
-        if: steps.python_poetry.outputs.enabled == 'true'
-        run: |
-          error_check=`(poetry env use 3.9 2>&1 || true)`
-          if ! grep -q "Please choose a compatible version" <<< $error_check
-          then
-            echo "PYTHON_VERSIONS=$(echo "${PYTHON_VERSIONS}" | jq -c '. + ["3.9"]')" >> $GITHUB_ENV
-          else
-            echo "Python 3.9 does not meet project requirements."
-          fi
-
-      - name: Set up Python 3.10
-        if: steps.python_poetry.outputs.enabled == 'true'
-        uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1 # v4
-        with:
-          python-version: "3.10"
-
-      - name: Check compatibility
-        if: steps.python_poetry.outputs.enabled == 'true'
-        run: |
-          error_check=`(poetry env use 3.10 2>&1 || true)`
-          if ! grep -q "Please choose a compatible version" <<< $error_check
-          then
-            echo "PYTHON_VERSIONS=$(echo "${PYTHON_VERSIONS}" | jq -c '. + ["3.10"]')" >> $GITHUB_ENV
-          else
-            echo "Python 3.10 does not meet project requirements."
-          fi
+          versions=()
+          while IFS= read -r version; do
```

Comment:
Less elegant than
```bash
for version in ${{ env.SUPPORTED_VERSIONS }}; do
```
but that's the price to pay for reusing the same configuration value in several places.

Reply:
Yes, but look at how much duplicate code this saves. It is super worth it